lib/workers/mixins/web_server_worker_mixin.rb

```diff
@@ -10,8 +10,6 @@ def do_work
 
   def do_before_work_loop
     @worker.release_db_connection
-
-    # Since thin traps interrupts, log that we're going away and update our worker row
```

Comment:
It might be a no-op of removing the at_exit and sending SIGTERM/SIGINT to the puma process.  If it logs the message from worker_base, we're getting the SignalException and logging and exiting correctly.